### PR TITLE
GUI/info game/21

### PR DESCRIPTION
**Pull request by GuyomT**
#33 in data_game.h
implementation of a main and a class InfoGame and GUI

### CODE REVIEW

File: src/zappy_gui/data_game.h [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    LEVEL5,
+    LEVEL6,
+    LEVEL7,
+    LEVEL8
+};
+typedef enum player_level_e player_level;
+
+/// \brief orientation of the player
+enum player_orientation_e{
+    ORIENTATIONUNDEFINED = 0,
```
Comment (LucasTesnier) [inline src/zappy_gui/data_game.h:39]: Same

File: src/zappy_gui/data_game.h [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    NORTH,
+    SOUTH,
+    EAST,
+    WEST
+};
+typedef enum player_orientation_e player_orientation;
+
+/// \brief status of the game
+enum game_status_e{
+    GAMESTATUSUNDEFINED = 0,
```
Comment (LucasTesnier) [inline src/zappy_gui/data_game.h:49]: Same

File: src/zappy_gui/data_game.h [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    PAUSE,
+    WIN,
+    LOOSE
+};
+typedef enum game_status_e game_status;
+
+/// \brief ressources of a player or of a tile
+struct ressource_s{
+    char *name;
```
Comment (LucasTesnier) [inline src/zappy_gui/data_game.h:58]: Need to add documentation for the struct member.

File: src/zappy_gui/data_game.h [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+//     {.name = "mendiane", .quantity = 0},
+//     {.name = "phiras", .quantity = 0},
+//     {.name = "thystame", .quantity = 0},
+//     {.name = "food", .quantity = 0},
+//     {.name = 0, .quantity = 0},
+// };
+
+/// \brief egg belonging to a team and with a life
+struct egg_s{
+    char *team;
```
Comment (LucasTesnier) [inline src/zappy_gui/data_game.h:76]: Same

File: src/zappy_gui/data_game.h [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    int life;
+};
+typedef struct egg_s egg_t;
+
+/// \brief player
+struct player_s {
+    int x;
+    int y;
+    int life;
```
Comment (LucasTesnier) [inline src/zappy_gui/data_game.h:85]: Same

File: src/zappy_gui/data_game.h [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    ressource_t *inventory;
+    char *team_name;
+    player_level level;
+    player_orientation orientation;
+};
+typedef struct player_s player_t;
+
+/// \brief tile
+struct tile_s {
+    int x;
```
Comment (LucasTesnier) [inline src/zappy_gui/data_game.h:97]: Same

File: src/zappy_gui/data_game.h [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+    int y;
+    ressource_t *ressources;
+    egg_t *eggs;
+};
+typedef struct tile_s tile_t;
+
+/// \brief data
+struct data_s {
+    player_t *players;
+    tile_t *tiles;
```
Comment (LucasTesnier) [inline src/zappy_gui/data_game.h:107]: Same

File: src/zappy_gui/main.cpp [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+int main(int ac, char **av)
+{
+    if (ac < 3)
+        std::cerr << "Not enough arguments" << std::endl;
+    if (ac > 3)
+        std::cerr << "Too many arguments" << std::endl;
+    try {
+        std::string machine(av[2]);
+        gui::GameInfo info = gui::GameInfo(std::stoi(av[1]), machine);
+    } catch (std::exception const &e) {
```
Comment (LucasTesnier) [inline src/zappy_gui/main.cpp:23]: Create a custom error class will be very usefull for logging informations for the next step of the project. Think about it

**Review by pariciadra — CHANGES_REQUESTED** [review]
Fix this tiny things pls

File: src/zappy_gui/GUI.hpp [hunk 9]
```diff
@@ -0,0 +1,40 @@
+/*
+** EPITECH PROJECT, 2022
+** ZAPPYMIRROR
+** File description:
+** GUI
+*/
+
+#ifndef GUI_HPP_
+#define GUI_HPP_
```
[the remark this answers is not in the document]
Comment (osvegn) [reply src/zappy_gui/GUI.hpp:9]: But not in cpp ;)

File: src/zappy_gui/GUI.cpp [hunk 10]
```diff
@@ -0,0 +1,50 @@
+/*
+** EPITECH PROJECT, 2022
+** ZAPPYMIRROR
+** File description:
+** GUI
+*/
+
```
[the remark this answers is not in the document]
Comment (osvegn) [reply src/zappy_gui/GUI.cpp:7]: The \file must be the path of the file:
src/zappy_gui/GUI.cpp

File: src/zappy_gui/GUI.hpp [hunk 11]
```diff
@@ -0,0 +1,46 @@
+/*
+** EPITECH PROJECT, 2022
+** ZAPPYMIRROR
+** File description:
+** GUI
+*/
+
+/// \file GUI class
```
Comment (osvegn) [inline src/zappy_gui/GUI.hpp:8]: \file must be the path of the file

Use a \brief if you want to describe the file

File: src/zappy_gui/GUI.hpp [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+#include <string>
+#include <vector>
+
+namespace gui {
+class GUI {
+   public:
+    GUI(size_t width, size_t height);
+    ~GUI();
+    /// \brief create the window with a name
+    void initWindow(std::string const &name = "ZappyGUI");
```
Comment (osvegn) [inline src/zappy_gui/GUI.hpp:24]: Need documentation for all parameters

Comment (osvegn) [reply src/zappy_gui/GUI.hpp:24]: \param

File: src/zappy_gui/GUI.hpp [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+    /// \brief close the window
+    void closeWindow();
+    /// \brief clear the screen
+    void clearScreen();
+    /// \brief display all what as been drawn on the window
+    void displayScreen();
+    /// \brief display the GUI
+    void displayGUI();
+   protected:
```
Comment (osvegn) [inline src/zappy_gui/GUI.hpp:33]: useless

File: src/zappy_gui/GameInfo.hpp [hunk 14]
```diff
@@ -0,0 +1,45 @@
+/*
+** EPITECH PROJECT, 2022
+** ZAPPYMIRROR
+** File description:
+** GameInfo
+*/
+
+/// \file Game Info class
```
Comment (osvegn) [inline src/zappy_gui/GameInfo.hpp:8]: \file must be the path of the file

File: src/zappy_gui/data_game.h [hunk 15]
```diff
@@ -0,0 +1,111 @@
...
+    TAKE_OBJECT,
+    SET_OBJECT,
+    DYING,
+    BROADCASTING
+};
+typedef enum player_status_e player_status;
+
+/// \brief level of the player
+enum player_level_e{
+    LEVELUNDEFINED = 0,
```
[the remark this answers is not in the document]
Comment (osvegn) [reply src/zappy_gui/data_game.h:38]: \brief all enum

File: src/zappy_gui/data_game.h (hunk 1 above)
Comment (osvegn) [reply src/zappy_gui/data_game.h:52]: \brief all enum

**Review by osvegn — CHANGES_REQUESTED** [review]
Some documentation is missing, please add it.
Don't forget to resolve conversation when you resolve a problem ;)

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Review by pariciadra — CHANGES_REQUESTED** [review]
Fix few things pls

**Review by osvegn — CHANGES_REQUESTED** [review]
Need some changes. Please resolve conflicts

**Comment by GuyomT** [comment]
Already have been implemented in another branch

**Pull request closed without merging**